### PR TITLE
Separate common flags and update flags

### PR DESCRIPTION
**Pull request by daisy-ycguo**
common flags are for all commands; update flags are for create and update commands.

### CODE REVIEW

File: pkg/core/root.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ func NewKnSourceCommand(knSourceFactory types.KnSourceFactory,
 
 	createCmd := commandFactory.CreateCommand()
 	addCommonFlags(knSourceParams, createCmd)
+	addUpdateFlags(knSourceParams, createCmd)
```
Comment (maximilien) [inline pkg/core/root.go]: why is the name not matching: `addUpdateFlags` and `createCmd`

Comment (maximilien) [reply pkg/core/root.go]: Wondering if we should have one for create even if its implementation is the same?

**Review by maximilien — APPROVED** [review]
See comment

File: pkg/core/root.go (hunk 1 above)
Comment (daisy-ycguo) [reply pkg/core/root.go]: actually, I think the update flags and create flags are same, for most situation.

**Review by maximilien — APPROVED** [review]
Thx

**Pull request merged**